pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -96,37 +96,50 @@ func machinesEqual(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
 	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
 }
 
+// isNodeJoin determines if a machine, in scope, should join of the cluster.
 func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachines []*clusterv1.Machine) (bool, error) {
 	switch set := scope.Machine.ObjectMeta.Labels["set"]; set {
 	case "node":
+		// Worker machines, not part of the controlplane, will always join the cluster.
 		return true, nil
 	case "controlplane":
+		// Controlplane machines will join the cluster if the cluster has an existing control plane.
+		controlplaneExists := false
+		var err error
 		for _, cm := range controlPlaneMachines {
 			m, err := actuators.NewMachineScope(actuators.MachineScopeParams{
-				Machine: cm,
-				Cluster: scope.Cluster,
-				Client:  a.clusterClient,
-				Logger:  a.log,
+				Machine:    cm,
+				Cluster:    scope.Cluster,
+				Client:     a.clusterClient,
+				Logger:     a.log,
+				AWSClients: scope.AWSClients,
 			})
 
 			if err != nil {
-				return false, errors.Wrapf(err, "failed to create machine scope for machine %q", cm.Name)
+				return false, errors.Wrapf(err, "failed to create machine scope for machine %q in namespace %q", cm.Name, cm.Namespace)
 			}
 
 			ec2svc := ec2.NewService(m.Scope)
 
-			ok, err := ec2svc.MachineExists(m)
+			controlplaneExists, err = ec2svc.MachineExists(m)
 			if err != nil {
-				return false, errors.Wrapf(err, "failed to verify existence of machine %q", m.Name())
+				return false, errors.Wrapf(err, "failed to verify existence of machine %q in namespace %q", m.Machine.Name, m.Machine.Namespace)
```

Comment:
Same as above, if namespace and machine is still needed with new logging this should use <namespace>/<name>